cmd/soroban-cli/src/strval.rs

```diff
@@ -438,7 +438,8 @@ impl Spec {
     /// May panic
     pub fn xdr_to_json(&self, val: &ScVal, output: &ScType) -> Result<Value, Error> {
         Ok(match (val, output) {
-            (ScVal::Map(None) | ScVal::Vec(None), ScType::Option(_)) => Value::Null,
+            (ScVal::Void, ScType::Val)
```

Comment:
Is there anything sensible we can do if `val != ScVal::Void && output == ScType::Val`? And wouldn't this still fail in that case?

Now that we have the value overhaul in, the below matchers seem to mostly be pretty trivial ScVal -> ScType "equality" checks... 🤔

Reply:
Most of them were before too.  The tricky part is still Enums, Unions, Maps, & Sets since they don't exist in ScVal.

Reply:
In the case of `val != ScVal::Void && output == ScType::Val` really it just means that we should just ignore the `ScType` and look only at the `ScVal`?  If so we should add back the `ScVal` --> `JSON` for this case.

Reply:
I'm saying, that really for `xdr_to_json`, the `output: &ScType` is sort of irrelevant. So yeah, for that case we should ignore the type, and just do `ScVal -> JSON`.

But more generally than that, I'm saying, maybe we should *always* ignore the type. The only cases where we do anything non-trivial are nulls, addresses, and results. 🤔